File: python/meteroid/models/batch_job_type.py
# this file is @generated
import enum


class BatchJobType(str, enum.Enum):
    EVENT_CSV_IMPORT = "EVENT_CSV_IMPORT"
    CUSTOMER_CSV_IMPORT = "CUSTOMER_CSV_IMPORT"
    SUBSCRIPTION_CSV_IMPORT = "SUBSCRIPTION_CSV_IMPORT"
    SUBSCRIPTION_PLAN_MIGRATION = "SUBSCRIPTION_PLAN_MIGRATION"
    TAX_REPORT_EXPORT = "TAX_REPORT_EXPORT"

    def __str__(self) -> str:
        return str(self.value)

File: go/batch_job_type.go
// this file is @generated
package meteroid

type BatchJobType string

// Known values of BatchJobType.
//
// The API may add new values over time; unknown values round-trip unchanged.
const (
	BatchJobTypeEventCsvImport            BatchJobType = "EVENT_CSV_IMPORT"
	BatchJobTypeCustomerCsvImport         BatchJobType = "CUSTOMER_CSV_IMPORT"
	BatchJobTypeSubscriptionCsvImport     BatchJobType = "SUBSCRIPTION_CSV_IMPORT"
	BatchJobTypeSubscriptionPlanMigration BatchJobType = "SUBSCRIPTION_PLAN_MIGRATION"
	BatchJobTypeTaxReportExport           BatchJobType = "TAX_REPORT_EXPORT"
)

// AllBatchJobTypeValues lists every BatchJobType value known to this SDK version.
var AllBatchJobTypeValues = []BatchJobType{
	BatchJobTypeEventCsvImport,
	BatchJobTypeCustomerCsvImport,
	BatchJobTypeSubscriptionCsvImport,
	BatchJobTypeSubscriptionPlanMigration,
	BatchJobTypeTaxReportExport,
}

// String returns the wire representation of the value.
func (e BatchJobType) String() string {
	return string(e)
}

// IsKnown reports whether the value is one this SDK version knows about.
func (e BatchJobType) IsKnown() bool {
	for _, known := range AllBatchJobTypeValues {
		if e == known {
			return true
		}
	}
	return false
}
